base/src/functions/logical.rs

```diff
@@ -166,94 +137,58 @@ impl Model {
                                 row,
                                 column,
                             }) {
-                                CalcResult::Boolean(value) => {
-                                    result = value || result;
-                                }
+                                CalcResult::Boolean(value) => result = Some(fold_fn(result, value)),
                                 CalcResult::Number(value) => {
-                                    if value != 0.0 {
-                                        return CalcResult::Boolean(true);
-                                    }
-                                }
-                                CalcResult::String(_value) => {
-                                    return CalcResult::Boolean(true);
+                                    result = Some(fold_fn(result, value != 0.0))
                                 }
                                 error @ CalcResult::Error { .. } => return error,
-                                CalcResult::Range { .. } => {}
-                                CalcResult::EmptyCell | CalcResult::EmptyArg => {}
+                                CalcResult::EmptyArg => debug_assert!(false, "unreachable"),
```

Comment:
This is not a bad idea at all, but this wold be the only place where we have a debug_assert! in the code.
Could you remove it and add a comment with the word "unreachable"?

Reply:
Yeah, no we can't panic. If there are more `unreachable!` in the codebase is a bug.
Let's just have a comment for the time being. I will most likely convert all those do debug_assert or something similar one day :)